### PR TITLE
Add pica library type definition files

### PR DESCRIPTION
**Pull request by hmtylmz**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

I write a type definition file and add pull request library itself. But author said:
https://github.com/nodeca/pica/pull/171

### CODE REVIEW

File: types/pica/pica-tests.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+let picaOptions: PicaOptions = { features: ['js', 'wasm', 'ww', 'cib'] };
+let resizer: Pica = new Pica(picaOptions);
```
Comment (peterblazejewicz) [inline types/pica/pica-tests.ts]: Just suggestions, as the options are optional, could we also include the parameterless constructor test as well? Exactly as documented on the package readme (all ctros are parameterless there):
```js
const pica = Pica();
pica.resize(img, canvas).then(...);
```
Thanks!

Comment (hmtylmz) [reply types/pica/pica-tests.ts]: Added tests with parameterless constructor

File: types/pica/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    unsharpRadius?: number;
+    // 0..255. Default = 0. Threshold for applying unsharp mask.
+    unsharpThreshold?: number;
+    // Promise instance. If defined, current operation will be terminated on rejection.
+    cancelToken?: string;
+}
+
+interface PicaResizeBufferOptions {
+    // Uint8Array with source data.
+    src?: number[];
```
Comment (peterblazejewicz) [inline types/pica/index.d.ts]: I believe this is required:
https://github.com/nodeca/pica/blob/master/README.md#resizebufferoptions---promise
It passed bits to 3rd party library:
https://github.com/nodeca/multimath#readme
Can you double-check this?
Thanks!

Comment (hmtylmz) [reply types/pica/index.d.ts]: Sorry. I fixed this.

**Review by peterblazejewicz — APPROVED** [review]
LGTM
Thanks for contribution!

**Review by andrewbranch — APPROVED** [review]

File: types/pica/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
+// Type definitions for pica 5.1
+// Project: https://github.com/nodeca/pica
+// Definitions by: Hamit YILMAZ <https://github.com/hmtylmz>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare class Pica {
```
Comment (wkritzinger) [inline types/pica/index.d.ts]: Shouldn't this be exported? I get `File '/node_modules/@types/pica/index.d.ts' is not a module.  TS2306` if nothing is exported from this file.

Comment (andrewbranch) [reply types/pica/index.d.ts]: Pica is a UMD module (meaning it works as a global, an AMD module, or a CommonJS module). So while it’s not exactly _wrong_ to simply declare it as a global, it’s incomplete. There should be an `export as namespace Pica` to indicate it’s available as UMD.

**Comment by namelesswc** [comment]
I'm having trouble using this.
As far as I understood, I don't need to import this like other packages `import * as Pica from "pica";`.
But directly using it like this `Pica.resize(image, canvas);` also doesn't work. I get `Property 'resize' does not exist on type 'typeof Pica'.`

**Comment by peterblazejewicz** [comment]
folks, I've made a RP to mitigate discussed problems:
#44858
Thanks for the patience!

**Pull request merged**